Subject: [PATCH] - Applies to e8b814d629a0c2073239828e63d50b125c013570of AMDSEV
diff --git a/common.sh b/common.sh
index 5f9da16..cb23a57 100755
--- a/common.sh
+++ b/common.sh
@@ -136,6 +136,8 @@ build_kernel()
 			run_cmd ./scripts/config --enable MLXSW_SPECTRUM_DCB
 			run_cmd ./scripts/config --module MLXSW_MINIMAL
 			run_cmd ./scripts/config --module MLXFW
+			run_cmd ./scripts/config --module SCSI_VIRTIO
+			run_cmd ./scripts/config --module VIRTIO_NET
 
 			run_cmd echo $COMMIT >../../source-commit.kernel.$V
 		popd >/dev/null
@@ -169,7 +171,7 @@ build_install_ovmf()
 		GCCVERS="GCC5"
 	fi
 
-	BUILD_CMD="nice build -q --cmd-len=64436 -DDEBUG_ON_SERIAL_PORT=TRUE -n $(getconf _NPROCESSORS_ONLN) ${GCCVERS:+-t $GCCVERS} -a X64 -p OvmfPkg/OvmfPkgX64.dsc"
+	BUILD_CMD="nice build -q --cmd-len=64436 -DDEBUG_ON_SERIAL_PORT=TRUE -n $(getconf _NPROCESSORS_ONLN) ${GCCVERS:+-t $GCCVERS} -a X64 -p OvmfPkg/AmdSev/AmdSevX64.dsc"
 
 	# initialize git repo, or update existing remote to currently configured one
 	if [ -d ovmf ]; then
@@ -194,12 +196,19 @@ build_install_ovmf()
 		run_cmd make -C BaseTools clean
 		run_cmd make -C BaseTools -j $(getconf _NPROCESSORS_ONLN)
 		. ./edksetup.sh --reconfig
-		run_cmd $BUILD_CMD
+		echo $BUILD_CMD
+		eval "$BUILD_CMD" || {
+			echo "ERROR: $BUILD_CMD"
+			echo "Attempting a second run" 
+			# If the $BUILD_CMD gives an error this might be fixed with:
+			touch OvmfPkg/AmdSev/Grub/grub.efi
+			# See: https://github.com/AMDESE/ovmf/issues/6 regarding:
+			# grub-mkimage: error: cannot open `/usr/lib/grub/x86_64-efi/linuxefi.mod': No such file or directory.
+			run_cmd $BUILD_CMD
+		}
 
 		mkdir -p $DEST
-		run_cmd cp -f Build/OvmfX64/DEBUG_$GCCVERS/FV/OVMF_CODE.fd $DEST
-		run_cmd cp -f Build/OvmfX64/DEBUG_$GCCVERS/FV/OVMF_VARS.fd $DEST
-		run_cmd cp -f Build/OvmfX64/DEBUG_$GCCVERS/FV/OVMF.fd $DEST
+		run_cmd cp -f Build/AmdSev/DEBUG_$GCCVERS/FV/OVMF.fd $DEST/DIRECT_BOOT_OVMF.fd
 
 		COMMIT=$(git log --format="%h" -1 HEAD)
 		run_cmd echo $COMMIT >../source-commit.ovmf
diff --git a/install.sh b/install.sh
index 3dd4d86..53f66a1 100755
--- a/install.sh
+++ b/install.sh
@@ -4,7 +4,7 @@
 
 # This will install all the dependent packages for qemu and ovmf to run
 if [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; then
-	apt-get -y install qemu ovmf
+	apt-get -y install qemu-system ovmf
 else
 	dnf install @virt edk2-ovmf -y
 fi
